js/src/elements/star-rating.ts

```diff
@@ -0,0 +1,289 @@
+const starRatingTemplate = document.createElement('template');
+starRatingTemplate.id = 'star-rating-template';
+
+/* TODO: Consider replacing with constructable stylesheets when there is broader support for Firefox and Safari.
+https://web.dev/custom-elements-v1/
+https://developer.mozilla.org/en-US/docs/Web/API/CSSStyleSheet */
+
+starRatingTemplate.innerHTML = `
+<style>
+	*,
+	::before,
+	::after {
+		box-sizing: border-box;
+	}
+	
+	:host {
+		--star-color: var(--theme-primary-active);
+	
+		display: block;
+	}
+
+
+	:host([disabled]) {	
+		--star-color: var(--theme-text);
+
+		cursor: not-allowed;
+	}
+	
+	fieldset {
+		display: flex;
+        margin: 0;
+        border: none;
+        padding-block: 0.75rem;
+	}
+	
+	#input-container {
```

Comment:
Is there a possibility that a page could have more than one rating widget? If so, should this be a class or other reusable selector?

Reply:
Updated. I tried using a class in the past and it wasn't filling the stars properly for whatever reason, but it seems to be fine now.

Reply:
This stylesheet and the element with the `#input-container` id are scoped to the element's shadow dom so in theory this is a fine practice and there should not be a collision risk.

Reply:
Ah, good deal - just my own misunderstanding of shadow DOM then! No worries!